### PR TITLE
feat(rtl): Initial RTL implementation with logical components and RTLCSS

### PR DESCRIPTION
**Pull request by Vignesh-Manogar-E3433**
Converted all CSS directional properties to logical properties. This was done using a PostCSS plugin. Included the plugin in this PR.

Known issues:
- Float: inline-start, inline-end does not have enough browser support. 

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)
 
## How Has This Been Tested?
This was tested in Chrome, Firefox, Safari.

### CODE REVIEW

File: packages/crayons-core/postcss-plugins/postcss-directional-to-logical.js [hunk 1]
```diff
@@ -0,0 +1,305 @@
+const postcss = require('postcss');
```
Comment (kaushal-freshworks) [inline packages/crayons-core/postcss-plugins/postcss-directional-to-logical.js:1]: This comment is for the dir structure. Can you create a folder called plugins under crayons-core and  move the postcss-plugins folder into this plugins folder?

**Review by kaushal-freshworks — CHANGES_REQUESTED** [review]
Liner needs to be added ?

**Comment by Vignesh-Manogar-E3433** [comment]
https://github.com/freshworks/crayons/pull/621#pullrequestreview-1003873670

@kaushal-freshworks , @srivalli-sivaramasankaran is working on the linter part. She will create a separate PR for linting.

File: packages/crayons-core/postcss-plugins/postcss-directional-to-logical.js (hunk 1 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/postcss-plugins/postcss-directional-to-logical.js:1]: Sure, will do.

Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/postcss-plugins/postcss-directional-to-logical.js]: Done.

File: packages/crayons-core/package.json [hunk 2]
```diff
@@ -83,6 +86,7 @@
     "plop": "^2.7.4",
     "postcss": "^8.3.9",
     "postcss-cli": "^9.0.1",
+    "postcss-directional-to-logical": "./postcss/postcss-directional-to-logical",
```
Comment (arvindanta) [inline packages/crayons-core/package.json:89]: I dont think we need this here as dependency. I see that its imported locally.

File: packages/crayons-i18n/package.json [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "@rollup/plugin-dynamic-import-vars": "^1.4.1",
     "@rollup/plugin-node-resolve": "^11.1.0",
     "@rollup/plugin-typescript": "^8.1.0",
+    "@types/estree": "^0.0.51",
```
Comment (arvindanta) [inline packages/crayons-i18n/package.json:41]: do we need this dependency here ?

File: packages/crayons-icon/package.json [hunk 4]
```diff
@@ -53,5 +53,8 @@
     "json-to-pretty-yaml": "^1.2.2",
     "meow": "^9.0.0",
     "svgo": "^1.3.2"
+  },
+  "devDependencies": {
+    "@types/estree": "^0.0.51"
```
Comment (arvindanta) [inline packages/crayons-icon/package.json:58]: do we need this dependency here ?

File: packages/crayons-core/package.json [hunk 5]
```diff
@@ -60,13 +60,16 @@
     "storybook": "run-p local storybook:start",
     "test:watch": "stencil test --spec --e2e --watchAll --verbose",
     "test": "stencil test --spec --e2e",
-    "test:ci": "stencil test --spec --ci --e2e --max-workers=2"
+    "test:ci": "stencil test --spec --ci --e2e --max-workers=2",
+    "directional-to-logical": "npx postcss ./src/components/**/*.scss ./src/components/**/**/**/*.scss --config ./plugins/postcss-plugins/postcss.config.js --replace --no-map --verbose"
```
Comment (arvindanta) [inline packages/crayons-core/package.json:64]: Can you pls reorganize this npm script alphabetically?

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-core/package.json (hunk 5 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/package.json]: Done.

File: packages/crayons-icon/package.json (hunk 4 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-icon/package.json:58]: Removed.

File: packages/crayons-i18n/package.json (hunk 3 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-i18n/package.json]: This was necessary when we had the plugin as a package. Removed this now.

File: packages/crayons-core/package.json (hunk 2 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/package.json]: Missed to remove this. Thanks. Have removed this now.

File: packages/crayons-icon/package.json (hunk 4 above)
Comment (arvindanta) [reply packages/crayons-icon/package.json:58]: I still see this dependency in crayons-icon package.json. Can you please check ?

Comment (Vignesh-Manogar-E3433) [reply packages/crayons-icon/package.json]: Yes, removed now.

**Review by kaushal-freshworks — APPROVED** [review]
LGTM

**Pull request merged**